tensorflow/src/TensorFlow/Internal/FFI.hs

```diff
@@ -187,8 +188,12 @@ createTensorData t = do
     -- Read data.
     len <- fromIntegral <$> Raw.tensorByteSize t
     bytes <- castPtr <$> Raw.tensorData t :: IO (Ptr Word8)
-    -- TODO(fmayle): Don't copy the data.
-    v <- S.fromList <$> peekArray len bytes
+    -- Note: We would like to avoid the copy by creating an S.Vector directly
+    -- from 'bytes' and calling Raw.deleteTensor when it gets GC'd, but we can't
+    -- because the vector may outlive the tensorflow session and become invalid.
+    mv <- SM.new len
```

Comment:
Nevermind, TensorData is using Word8. The fromIntegral part is still suspect though as it seems to be :: CULong -> Int.